### PR TITLE
Export types and classes needed

### LINKED ISSUE #63: Ability to create custom resources
We're looking for the ability to create a custom useQuery function that provides default params that we would use everywhere. Right now it seems that everything we would need to do that is private. 

What are your thoughts on this @josemarluedke ?

### PR DESCRIPTION
**Pull request by bakerac4**
Fixes #63 and #58

### CODE REVIEW

**Comment by josemarluedke** [comment]
Hi @bakerac4! 

We currently export a few type utilities that allow us to make custom queries / pre-defined queries. Exports are here: https://github.com/josemarluedke/glimmer-apollo/blob/main/packages/glimmer-apollo/src/-private/usables.ts

Wouldn't that be enough? 

Here is an usage example:

```ts
import { gql, useQuery } from 'glimmer-apollo';
import type { UseQuery } from 'glimmer-apollo';
import type {
  MyQuery,
  MyQueryQueryVariables
} from './my-graphql-types';

export function useMyQuery<
  T extends UseQuery<
    MyQuery,
   MyQueryQueryVariables 
  >
>(ctx: Object, args?: T['args']): T['return'] {
  return useQuery(ctx, () => [
    gql`
      query MyQuery($id: ID!) {
        myQuery(id: $id) {
          id
        }
      }
    `,
    args ? args() : {}
  ]);
}
```

**Comment by josemarluedke** [comment]
@bakerac4 I love the custom resource tests, can you rework that to use the type utilities I mentioned above?

**Comment by bakerac4** [comment]
> 

I believe that fixes part of the issue with types. But for our generic query that sets defaults, you don't have the query type.

```export function useMyQuery<
  T extends UseQuery<
    **GenericQuery**,
   **GenericQueryVariables** 
  >
>(ctx: Object, query: **WhatsThisType**, args?: T['args']): T['return'] {
  return useQuery(ctx, () => [
    query,
    args ? args() : {}
  ]);
}

```

File: packages/glimmer-apollo/src/index.ts [hunk 1]
```diff
@@ -11,3 +11,15 @@ export type {
   UseMutation,
   UseSubscription
 } from './-private/usables';
+
+export { useResource } from './-private/use-resource';
+export {
```
Comment (josemarluedke) [inline packages/glimmer-apollo/src/index.ts:16]: Let's export this as type.
```suggestion
export type {
```

File: packages/glimmer-apollo/src/index.ts [hunk 2]
```diff
@@ -11,3 +11,15 @@ export type {
...
+  QueryOptions,
+  QueryResource,
+  QueryPositionalArgs
+} from './-private/query';
+export {
```
Comment (josemarluedke) [inline packages/glimmer-apollo/src/index.ts:21]: Same here.

File: packages/glimmer-apollo/src/index.ts [hunk 3]
```diff
@@ -11,3 +11,14 @@ export type {
...
+export type {
+  QueryOptions,
+  QueryResource,
+  QueryPositionalArgs
+} from './-private/query';
+export type {
+  MutationOptions,
+  MutationResource,
+  MutationPositionalArgs
+} from './-private/mutation';
```
[the remark this answers is not in the document]
Comment (bakerac4) [reply packages/glimmer-apollo/src/index.ts:23]: Good call. Completely forgot about that. Should be all set.

**Comment by josemarluedke** [comment]
@bakerac4 Thanks for working on this.

**Comment by josemarluedke** [comment]
Released in v0.5.1.

**Pull request merged**